src/components/calcite-pick-list/calcite-pick-list.tsx

```diff
@@ -193,6 +196,31 @@ export class CalcitePickList {
     });
   }
 
+  handleFilter = (event) => {
+    const filteredData = event.detail;
+    const values = filteredData.map((item) => item.value);
+    this.items.forEach((item) => {
+      if (values.indexOf(item.value) === -1) {
+        item.setAttribute("hidden", "");
+      } else {
+        item.removeAttribute("hidden");
+      }
+    });
+  };
+
+  getItemData(): Record<string, string>[] {
+    const result: Record<string, string>[] = [];
+    this.items.forEach((item) => {
+      const obj: Record<string, string> = {};
+      Array.from(item.attributes).forEach((attr) => {
+        obj[attr.name] = attr.value;
+      });
+      obj.metadata = item.metadata;
```

Comment:
TypeScript: src/components/calcite-pick-list/calcite-pick-list.tsx:218:7
           Type 'object' is not assignable to type 'string'.

    L217:  });
    L218:  obj.metadata = item.metadata;
    L219:  result.push(obj);